### PR TITLE
[FEATURE] Add descriptions for all Composer scripts

### LINKED ISSUE #1079: Add descriptions for our Composer commands
https://getcomposer.org/doc/articles/scripts.md#custom-descriptions-

### PR DESCRIPTION
**Pull request by oliverklee**
Composer will list all commands with their descriptions when run
without any arguments.

Fixes #1079

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -99,6 +99,25 @@
...
+        "ci": "Runs all dynamic and static code checks.",
+        "ci:composer:normalize": "Checks the formatting and structure of the composer.json.",
+        "ci:dynamic": "Runs all dynamic tests (i.e., currently the unit tests).",
+        "ci:php:fixer": "Checks the code stile with PHP CS Fixer.",
+        "ci:php:lint": "Lints the PHP files for syntax errors.",
+        "ci:php:md": "Checks the code size limits with PHPMD.",
+        "ci:php:psalm": "Checks the typing with Psalm.",
+        "ci:php:sniff": "Checks the code style with PHP_CodeSniffer.",
+        "ci:static": "Runs all static code analysis checks for the code and the composer.json.",
+        "ci:tests": "Runs all dynamic tests (i.e., currently the unit tests).",
```
Comment (JakeQZ) [inline composer.json:112]: Not sure how this was intended, and perhaps beyond the scope of this PR, but if `ci:dynamic` and `ci:tests` have the same description and cannot be distinguished, we should remove one or other.

**Review by JakeQZ — CHANGES_REQUESTED** [review]
There's one typo, and a few clarity issues.

File: composer.json [hunk 2]
```diff
@@ -99,6 +99,25 @@
         "psalm:baseline": "@php \"./.phive/psalm\" --set-baseline=psalm.baseline.xml",
         "psalm:cc": "@php \"./.phive/psalm\" --clear-cache"
     },
+    "scripts-descriptions": {
+        "ci": "Runs all dynamic and static code checks.",
+        "ci:composer:normalize": "Checks the formatting and structure of the composer.json.",
```
[the remark this answers is not in the document]
Comment (oliverklee) [reply composer.json:104]: No, as the description usually is output on the console anyway, where we have a monospace font.

File: composer.json (hunk 1 above)
Comment (oliverklee) [reply composer.json:112]: Good point. I have added #1094 for this.

**Comment by JakeQZ** [comment]
On 16/06/2021 20:08, Oliver Klee wrote:
>
> @oliverklee requested your review on: #1092 [FEATURE] Add descriptions for all Composer scripts.

Did you forget to `push` anything?

File: composer.json [hunk 3]
```diff
@@ -99,6 +99,25 @@
...
+        "ci:dynamic": "Runs all dynamic tests (i.e., currently the unit tests).",
```
[the remark this answers is not in the document]
Comment (JakeQZ) [reply composer.json:105]: I seem to remember a previous discussion regarding a comma after "i.e.".  [It's been discussed elsewhere](https://english.stackexchange.com/questions/6894/should-there-be-a-comma-after-i-e).

To me, a comma works only to separate "currently" as some kind of additional clause (the linguistic name of which I'm not sure) and feels optional, but if used should be on both sides (like if you put "currently" in brackets).

**Comment by oliverklee** [comment]
I indeed forgot to push. Rebased and pushed now.

File: composer.json [hunk 4]
```diff
@@ -99,6 +99,25 @@
         "psalm:baseline": "@php \"./.phive/psalm\" --set-baseline=psalm.baseline.xml",
         "psalm:cc": "@php \"./.phive/psalm\" --clear-cache"
     },
+    "scripts-descriptions": {
+        "ci": "Runs all dynamic and static code checks.",
+        "ci:composer:normalize": "Checks the formatting and structure of the composer.json.",
+        "ci:dynamic": "Runs all dynamic tests (i.e. currently, the unit tests).",
```
Comment (JakeQZ) [inline composer.json:105]: I think there should either be a comma each side of "currently", or no comma either side, but not one on one side but not the other.

File: composer.json [hunk 5]
```diff
@@ -99,6 +99,25 @@
...
+        "ci:php:fixer": "Checks the code style with PHP CS Fixer.",
+        "ci:php:lint": "Lints the PHP files for syntax errors.",
+        "ci:php:md": "Checks the code complexity with PHPMD.",
+        "ci:php:psalm": "Checks the types with Psalm.",
+        "ci:php:sniff": "Checks the code style with PHP_CodeSniffer.",
+        "ci:static": "Runs all static code analysis checks for the code and the composer.json.",
+        "ci:tests": "Runs all dynamic tests (i.e., currently the unit tests).",
```
Comment (JakeQZ) [inline composer.json:112]: This has the same description so should also be changed consistently.

**Review by JakeQZ — CHANGES_REQUESTED** [review]
I think "currently" can be separated with commas either side, and they are kind-of optional, but if omitted both should be omitted.  Also, this description appears twice - sorry I did not flag that in the original review.

Other corrections and clarifications are fine.

File: composer.json (hunk 4 above)
Comment (oliverklee) [reply composer.json]: Change in both places and repushed.

**Review by JakeQZ — APPROVED** [review]

**Pull request merged**